### ISSUE TITLE
Addressing Clickbait

### ISSUE THREAD

**Issue by Quotatron3000** [ajayyy/SponsorBlock#1211]
Misleading clickbait is no new phenomena on YouTube, however it's unclear (bug? Clarification on tools usage?) how misleading clickbait fits within the application of SponsorBlock. It's clear SponsorBlock is designed to avoid people's time being wasted by fluff and filter in videos, but misleading clickbait videos are _entirely_ fluff and filler as they never fulfill the original title.

Hypothetically you could set a 'filler tangent' to the entire thing but it feels like it is abusing SponsorBlock. At the same time, I don't want other users to waste their time getting caught up on a video that never offers up the goods it promises in both thumbnail and title. 

A really good example of this is: https://www.youtube.com/watch?v=4IcVIF3_708
The title (at time of writing) reads: _"James Webb Telescope Terrifying Discovery At The Edge Of The Universe!"_ but all it talks about is the specifications of the telescope, and universe expansion, and never once mentions what is so "terrifying" or what has been "discovered". Do users label the entire thing with filler tangent (UI clarification), or should there be a new feature added to label the video as clickbait?

I notice misleading clickbait videos have no highlight marker on them, but they're nearly impossible to distinguish from videos that haven't had segments applied to them, or have only had partial SponsorBlock segments added. And how do you distinguish filler tangent if the entire video is basically filler tangent that doesn't go anywhere?

It'd be nice if there was a little yellow mark before clicking a video (superimposed on the thumbnail) indicating it's likely clickbait, although I feel a variation of the 'highlight' feature called 'it's clickbait' that simply skips to the end of the video would suffice as well. You may want to distinguish between title clickbait (where the title is misleading) and thumbnail clickbait (where the title is true, but the thumbnail never appears in the video so it cannot be highlighted).

**Comment by ajayyy** [comment]
Check out the discussion on #clickbait on Discord/Matrix.

I don't believe a video can be "clickbait" or "not clickbait". I would prefer to solve the issue but replacing titles/thumbnails to not be misleading.

**Comment by ajayyy** [comment]
To clarify, nothing in the current feature set is meant to do what you are looking for. You are not supposed to label an entire video as Filler.

**Comment by Quotatron3000** [comment]
> For my use, clickbait a neutral term meaning something that is attempting to attract clicks

Okay, I think picking a clarifying term like 'misleading title/thumbnail' would make it clear what it's for, given as we have different views on 'clickbait', other users would too and it is better to be explicit.

>To clarify, nothing in the current feature set is meant to do what you are looking for. You are not supposed to label an entire video as Filler.

I got that impression which is why I didn't use it like that. Essentially aiming for some way to communicate to SponsorBlock users that 'a highlight has been searched for and wasn't found (or is not applicable)'. This would also make it easier to distinguish new, unmarked videos from ones that have been properly searched but a highlight wasn't found.

Otherwise every SponsorBlock user will encounter a highlight-less video, and might opt to look for a highlight that isn't there. I suppose in a way this expands beyond misleading videos as a video might have a truthful title and creative thumbnail where neither technically appear in the video.

**Comment by ccuser44** [comment]
I really don't see why we need a clickbait category anyway.

**Comment by RubenKelevra** [comment]
@Quotatron3000 that's a pretty neat idea as those videos usually get a lot of clicks they are recommended no matter what.

I feel like we might want to hide the original title and replace it with "hidden video title" or something generic to mark them as "not containing the content which is described in the title", to avoid confusion with the word “misleading” for general misinformation/conspiracy theories etc. which IMHO should not be included in this, as the title is not the issue here.

We might want to do the same with the thumbnails, by just replacing them with a grey box and a crossed eye symbol or something like this.

We must just make sure that we hash the title/picture and store it along with the mark, to allow the creators to improve the by correcting the title. :)

**Comment by cwille97** [comment]
Had a few ideas related to this:

- Measure clickbait as a % rating. Each user could mark a video as clickbait or not and then each video displays the % of users who rate it as clickbait (after a certain minimum to avoid small samples skewing the rating)
- Crowd sourced TL;DR (too long, didn't read). I envision this as a comment section within the SponsorBlock UI which allows users to write a summary of the video they watched and vote on the best summary. The top-voted summary is displayed in the main YouTube UI alongside the description.

I also like the idea others have mentioned regarding rewording a "clickbaity" title, but I think the TL;DR or summary is a nice addition to this because even if the title is not clickbait, you may want a bullet point list of topics to decide if the video is worth your time. This is something that some creators do using timestamps, and sometimes people will comment a summary of the main points in the YouTube comments. It'd be nice to have a crowd sourced version of this in SponsorBlock since it's not always present in the YouTube comments, and there's an obvious bias in allowing creators to leave their own summary.

**Comment by ajayyy** [comment]
> This is something that some creators do using timestamps

Could be useful to show the chapters in a video on hover, especially once #1001 is done

**Comment by cwille97** [comment]
I like that, probably provides 80% of the info of a TLDR without all of the heavy-handedness of having to maintain a database of comments (and the problems associated with that such as filtering for spam, phishing, extreme profanity).

**Comment by farOverNinethousand** [comment]
Not a bad request but I disagree with the general "label as clickbait" idea because:
- how to define where clickbait begins and where it ends
- often nowdays, creators may rename videos later which, compared to the in-video segments voting is more likely to destroy already submitted SponsorBlock clickbait votes (can YT videos be edited later on without the videoID changing? I guess if the answer to this is yes, still content-creators are rarely doing this.)
- how to do a proper "clickbait percentage" rating? I don't see a way to find a general solution here!

For me by far the best idea would be to call this "misleading title" and allow users to submit a "better title" which can then be displayed below the original one or alternatively, replace it entirely.
The entire feature could as well be called "Better title", no need to mention the word "clickbait" here if you ask me.

Problems I see on the technical side:
While SponsorBlock atm only works on/for single videos upon when the user wants to access them, this feature would be most effective to also be used on search-results/playlists or in general, lists of videos.
This would on one side make the implementation more complex and on the 2nd side add a lot more client side API requests(?)
Please correct me if I'm wrong.

**Comment by Quotatron3000** [comment]
I've thought more in-depth about the proposal.

I like SponsorBlock's pre-existing tagging system where videos can be tagged entirely (for example, showing a video is one whole sponsorship segment).

In terms of video titles changing, I think this could be solved by ensuring something like 'misleading title' only applies if the video title still matches a lowercase copy of the title. The title is only stored if the video is flagged as misleading. If the title changes, then the tag no longer shows.

If we're including misleading thumbnails, then a basic checksum hash on the image can be stored (rather than the image itself), and if the checksum changes, then the tag no longer applies. If the video is tagged misleading image, the checksum is done once by the server, then the client does their own checksum locally to compare to what the server has stored.

Although it would be nice for a video to show a tag in the thumbnail before you click on it, I agree that would be API intensive, and having the tag on the video itself would be sufficient, as it is only really the watching of the video that wastes time, rather than the clicking on it.

In terms of writing alternative titles, I think there's a high risk of abuse. It might be worthwhile trying to standardise what is misleading rather than customising titles, for example 'image not in the video', 'claim doesn't happen in the video', 'doesn't answer question', 'exaggerates claims'. It would always be possible to devise more. So you might have an option to tag as misleading, a dropdown to pick the reason(s) why, or an option to submit a suggested new category.

I don't think such a system should be used to police truth, per se, but highlight when a video title or image contradicts what's actually in the video itself. So for example, if a video shows Luke Skywalker and says 'Luke Skywalker's best skills' (a fictitious character) but the video is about Darth Vader (another fictitious character), that would be a contradiction, even though it's made up and not real.

On a side-note, what could be interesting is a complete spin-off from SponsorBlock that uses crowdsourcing to tag and classify video types (EG music, educational, entertainment, lets plays, etc), to have a searchable service that says 'here are all the educational videos, music videos, gaming videos', etc. Just a passing thought.

**Comment by ajayyy** [comment]
The plan right now is to _replace_ titles and thumbnails, and we are finding examples of good and bad exmaples to figure out some detailed guidelines

**Comment by farOverNinethousand** [comment]
Where are you collecting those examples?
As always, LTT videos are great "bad examples" e.g.:
https://www.youtube.com/watch?v=aKkVqmvs4NA
Title by the time I wrote this post: "I Let My Wife Down..."

**Comment by ajayyy** [comment]
On Discord/Matrix.

I don't think LTT has one good example...

**Comment by farOverNinethousand** [comment]
ThxYes LTT has always been the perfect example to justify the existence of SponsorBlock :) Am 29.04.2022 09:11 schrieb Ajay Ramachandran ***@***.***>:
On Discord/Matrix.
I don't think LTT has one good example...

**Comment by RubenKelevra** [comment]
Quotatron3000 wrote:
> Although it would be nice for a video to show a tag in the thumbnail before you click on it, I agree that would be API intensive, and having the tag on the video itself would be sufficient, as it is only really the watching of the video that wastes time, rather than the clicking on it.

I mean the browser could hold a local database for the video-ids + checksums for title and thumbnail plus a way to ask the server for updates from time to time.

This avoids asking the API for every single video in a long listing, and also preserves a bit more privacy – this behavior would be similar to an adblocker which fetches updated lists from time to time.

This also greatly reduces the response time for applying the changes inside the browser.

If the video is loaded anyway, we could ask the user if he agrees on a slider how "clickbaity" the title/thumbnail is (in which case we would need to show it).

Additionally, we could also show the title/thumbnail at the end of a video and let the users rate there as well. This would increase the amount of ratings, as you're probably not willing to go back to the history of YouTube to look at the thumbnail after a longer video.

Since all users submit just their opinion, we can do a geometric mean over the 95th percentile.

**Comment by farOverNinethousand** [comment]
@RubenKelevra 
I like the idea to ask the user for title feedback at the end of each video.
Also regarding API requests:
Of course when e.g. viewing search-results, API could allow to request such information for a comma separated list of videoIDs so not so many requests are needed.

**Comment by RubenKelevra** [comment]
@farOverNinethousand wrote:

> Of course when e.g. viewing search-results, API could allow to request such information for a comma separated list of videoIDs so not so many requests are needed.

Yeah sure, wasn't really thinking of a single listing, but if you scroll and the browser is loading more videos from YouTube on demand. So just scrolling will do additional API calls.

**Comment by Quotatron3000** [comment]
> I mean the browser could hold a local database for the video-ids + checksums for title and thumbnail plus a way to ask the server for updates from time to time.
> 
> This avoids asking the API for every single video in a long listing, and also preserves a bit more privacy – this behavior would be similar to an adblocker which fetches updated lists from time to time.
> 
> This also greatly reduces the response time for applying the changes inside the browser.

I think a local caching approach would be a really good idea to reduce overheads, the data could even be compressed on sending up/down to reduce bandwidth even further. Could aggregate it into channel groups, as popular channels will likely be prompted for more frequent updates than less popular ones, so instead of sending down a large chunk of all possible channels everywhere every single time, you're only sending down the chunks of relevant updates.

Having the local cache means you could have the video thumbnails display a hint as it's not hitting the main server. Maybe a fade out on the text/image to make it less prominent (prominence could even be proportionate to the slide rule you're thinking of). I could also see locally hosted alternate video titles (solving the moderation issue as only the user sees their own titles), but I think the value comes from the community benefit of shared effort.

I think in general SponsorBlock does need a sort of tagging approach. I think as SB gains popularity as the next major adblock tool, video producers will try to further intertwine their works with their advertising making timestamps more difficult to use appropriately.

There was one video by Tom Scott (https://www.youtube.com/watch?v=uXlQuTRSmzc) where the whole point of the video is his robot double does an advert, which a lot of SB submitters have blocked out large chunks of as sponsorship (rather than tagging the entire thing as a sponsorship).

**Comment by ajayyy** [comment]
About the Tom scott video, the current result is the compromise we settled on after discussions on discord

**Comment by RubenKelevra** [comment]
> I think a local caching approach would be a really good idea to reduce overheads, the data could even be compressed on sending up/down to reduce bandwidth even further. Could aggregate it into channel groups, as popular channels will likely be prompted for more frequent updates than less popular ones, so instead of sending down a large chunk of all possible channels everywhere every single time, you're only sending down the chunks of relevant updates.

Well, I thought more about delta updates. The database could just "write out" all changes and offer hourly updates. So the browser would just fetch the hourly updates and apply them to its local database.

And no, I don't think there should be any preferences for any kind of channel size etc. all or nothing. :)

**Comment by ccuser44** [comment]
Theres already an extensiin that does [Clickbait Remover for Youtube](https://github.com/pietervanheijningen/clickbait-remover-for-youtube/) - [Firefox Download](https://addons.mozilla.org/en-US/firefox/addon/clickbait-remover-for-youtube/) - [Chrome Download](https://chrome.google.com/webstore/detail/clickbait-remover-for-you/omoinegiohhgbikclijaniebjpkeopip)

**Comment by iodream** [comment]
@ccuser44 I believe all the extension does is replace the thumbnail with a frame from the video and only removes capitalization from the title, which isn't a real solution imo. Some of the ideas proposed here would have far more impact.

**Comment by ajayyy** [comment]
As an update on the title and thumbnail replacement idea, I'm working on it and the it is pretty far along

**Comment by ccuser44** [comment]
> @ccuser44 I believe all the extension does is replace the thumbnail with a frame from the video and only removes capitalization from the title, which isn't a real solution imo. Some of the ideas proposed here would have far more impact.

It does solve the issue created by clickb8 thumbnails. And mitigates nearly all of the clickbait power off the video title.

Also its highly subjective to think what constitutes as clickbait.

**Comment by ajayyy** [comment]
https://DeArrow.ajay.app
